### PR TITLE
SNOW-749412: snowpark python functions sample code batch 6

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/functions.py [hunk 1]
```diff
@@ -1173,7 +1204,21 @@ def degrees(e: ColumnOrName) -> Column:
 
 
 def radians(e: ColumnOrName) -> Column:
-    """Converts degrees to radians."""
+    """Converts degrees to radians.
+
+    Examples::
+
+        >>> df = session.create_dataframe([[1.111], [2.222], [3.333]], schema=["a"])
+        >>> df.select(radians(col("a")).alias("result")).show()
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/functions.py:1212]: Could we use a `CAST` to limit the number of digits after the floating point here?

File: src/snowflake/snowpark/functions.py [hunk 2]
```diff
@@ -1285,21 +1330,65 @@ def ltrim(e: ColumnOrName, trim_string: Optional[ColumnOrName] = None) -> Column
 
 def rpad(e: ColumnOrName, len: Union[Column, int], pad: ColumnOrName) -> Column:
     """Right-pads a string with characters from another string, or right-pads a
-    binary value with bytes from another binary value."""
+    binary value with bytes from another binary value.
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/functions.py:1333]: Could we add "`e` is padded to length `len` with characters/bytes from `pad`".

File: src/snowflake/snowpark/functions.py [hunk 3]
```diff
@@ -1419,7 +1520,21 @@ def pow(
 
 def round(e: ColumnOrName, scale: Union[ColumnOrName, int, float] = 0) -> Column:
     """Returns values from the specified column rounded to the nearest equal or
-    smaller integer."""
+    smaller integer.
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/functions.py:1523]: This is not true, 1.5 is rounded to 2 by default, the [Snowflake round function](https://docs.snowflake.com/en/sql-reference/functions/round) has a third parameter `mode` that can be either [half away from zero](https://en.wikipedia.org/wiki/Rounding#Rounding_half_away_from_zero) or [half to even](https://en.wikipedia.org/wiki/Rounding#Rounding_half_to_even). The default is rounding half away from zero.

File: src/snowflake/snowpark/functions.py [hunk 4]
```diff
@@ -3019,28 +3220,112 @@ def array_to_string(array: ColumnOrName, separator: ColumnOrName) -> Column:
...
 
 def object_construct(*key_values: ColumnOrName) -> Column:
-    """Returns an OBJECT constructed from the arguments."""
+    """Returns an OBJECT constructed from the arguments.
+
+    Example::
+
+        >>> from snowflake.snowpark.types import StructType, StructField, VariantType, StringType
+        >>> df = session.create_dataframe(
+        ...     [["name", "Joe"], ["zip", "98004"]],
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/functions.py:3255]: Could we have a row with `NULL` key and another row with `NULL` value? So the difference between `object_construct` and `object_construct_keep_null` is obvious?

File: src/snowflake/snowpark/functions.py [hunk 5]
```diff
@@ -3019,28 +3220,112 @@ def array_to_string(array: ColumnOrName, separator: ColumnOrName) -> Column:
...
 def object_construct_keep_null(*key_values: ColumnOrName) -> Column:
     """Returns an object containing the contents of the input (i.e. source) object with one or more
-    keys removed."""
+    keys removed.
+
+    Example::
+
+        >>> from snowflake.snowpark.types import StructType, StructField, VariantType, StringType
+        >>> df = session.create_dataframe(
+        ...     [["key_1", "one"], ["key_1", None]],
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/functions.py:3283]: Ditto.

File: src/snowflake/snowpark/functions.py [hunk 6]
```diff
@@ -3019,28 +3220,112 @@ def array_to_string(array: ColumnOrName, separator: ColumnOrName) -> Column:
...
+    The input key must not exist in the object.
+
+    Example::
+
+        >>> df = session.sql(
+        ...     "select object_construct(a,b,c,d,e,f) as obj, k, v from "
+        ...     "values('age', 21, 'zip', 21021, 'name', 'Joe', 'age', 0),"
+        ...     "('age', 26, 'zip', 94021, 'name', 'Jay', 'age', 0) as T(a,b,c,d,e,f,k,v)"
+        ... )
+        >>> df.select(object_delete(col("obj"), col("k")).alias("result")).show()
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/functions.py:3314]: There seems to be no keys removed here? Could we show a case where `keys` is not empty?

Comment (sfc-gh-aling) [reply src/snowflake/snowpark/functions.py]: I've updated the sample code to make it explicitly that key "age" is removed

File: src/snowflake/snowpark/functions.py (hunk 5 above)
Comment (sfc-gh-aling) [reply src/snowflake/snowpark/functions.py]: nice catch, updated

File: src/snowflake/snowpark/functions.py (hunk 4 above)
Comment (sfc-gh-aling) [reply src/snowflake/snowpark/functions.py]: sure, code updated

File: src/snowflake/snowpark/functions.py (hunk 3 above)
Comment (sfc-gh-aling) [reply src/snowflake/snowpark/functions.py]: good finding, I removed `nearest equal or smaller integer.` description, I think just using `rounded` should be good

File: src/snowflake/snowpark/functions.py (hunk 1 above)
Comment (sfc-gh-aling) [reply src/snowflake/snowpark/functions.py]: updated, how does it look to you now?

File: src/snowflake/snowpark/functions.py (hunk 2 above)
Comment (sfc-gh-aling) [reply src/snowflake/snowpark/functions.py]: added

**Review by sfc-gh-stan — APPROVED** [review]
LGTM, thank you Adam!

File: src/snowflake/snowpark/functions.py [hunk 7]
```diff
@@ -4096,8 +4532,24 @@ def ntile(e: Union[int, ColumnOrName]) -> Column:
     Divides an ordered data set equally into the number of buckets specified by n.
     Buckets are sequentially numbered 1 through n.
 
-    Args:
-        e: The desired number of buckets; must be a positive integer value.
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/functions.py:4100]: why are we removing arg definition?

Comment (sfc-gh-aling) [reply src/snowflake/snowpark/functions.py:4100]: it must be removed by mistake, let me add it back

**Review by sfc-gh-sfan — APPROVED** [review]

**Pull request merged**